src/libexpr/primops.cc

```diff
@@ -3505,6 +3505,50 @@ static RegisterPrimOp primop_mapAttrs({
     .fun = prim_mapAttrs,
 });
 
+static void prim_filterAttrs(EvalState & state, const PosIdx pos, Value ** args, Value & v)
+{
+    state.forceAttrs(*args[1], pos, "while evaluating the second argument passed to builtins.filterAttrs");
+
+    if (args[1]->attrs()->empty()) {
+        v = *args[1];
+        return;
+    }
+
+    state.forceFunction(*args[0], pos, "while evaluating the first argument passed to builtins.filterAttrs");
+
+    auto attrs = state.buildBindings(args[1]->attrs()->size());
+
+    for (auto & i : *args[1]->attrs()) {
+        Value * vName = Value::toPtr(state.symbols[i.name]);
+        Value * vFun2 = state.allocValue();
+        vFun2->mkApp(args[0], vName);
+        Value res;
+        state.callFunction(*vFun2, *i.value, res, noPos);
```

Comment:
Is this `mkApp()` needed? You can call `callFunction()` with multiple arguments, which would save an allocation.